app/views/vm_common/_snap.html.haml

```diff
@@ -1,40 +1,30 @@
-= form_tag({:action => 'snap_vm',
-            :id     => @record},
-           :id     => "vm_snap_form",
-           :method => :post) do
-  = render :partial => "layouts/flash_msg"
-  #form_div
-    %h3= _('Snapshot Information')
-    .form-horizontal
-      %div{:class => "form-group #{@record.vendor == "redhat" ? "hidden" : ""}"}
-        %label.control-label.col-md-2
-          = _('Name')
-        .col-md-8
-          = text_field_tag("name", @name, :maxlength => ViewHelper::MAX_NAME_LEN, :class => "form-control", "data-miq_focus" => true)
-      .form-group
-        %label.control-label.col-md-2
-          = _('Description')
-        .col-md-8
-          = text_area_tag("description", @description, :size => "50x4")
-      %div{:class => "form-group #{@show_snapshot_memory_checkbox ? "" : "hidden"}"}
-        %label.control-label.col-md-2
-          = _('Snapshot VM memory')
-        .col-md-8
-          = check_box_tag("snap_memory", "1", false)
-  - unless @edit
-    %hr
-    %table{:width => "100%"}
-      %tr
-        %td{:align => "right"}
-          = button_tag(_("Create"),
-            :name  => "create",
-            :class => "button",
-            :alt   => (t = _("Create snapshot")),
-            :title => t,
-            :type  => "submit")
-          = button_tag((t = _("Cancel")),
-            :name  => "cancel",
-            :class => "button",
-            :alt   => t,
-            :title => t,
-            :type  => "submit")
+#snap-form
+:javascript
+  var snapForm = ManageIQ.component.newInstance(
+    'SnapshotForm',
+    {
+      onSubmit: function(values) {
+        if(values.snap_memory) {
+          values.snap_memory = 1;
+        }
+        miqAjaxButton('#{url_for_only_path(:action => "snap_vm", :id => @record.id, :button => @edit ? "create" : nil)}', values);
```

Comment:
why split the logic? and have it inline? I would expect that if we move something to react, we don't use haml anymore?